controllers/vendors.go

```diff
@@ -50,20 +51,24 @@ func VendorFlashSystemCluster() odfv1alpha1.StorageKind {
 func (r *StorageSystemReconciler) isVendorSystemPresent(instance *odfv1alpha1.StorageSystem, logger logr.Logger) error {
 
 	var err error
+	var relatedObject runtime.Object
 
 	if instance.Spec.Kind == VendorStorageCluster() {
 		logger.Info("get storageCluster")
 		storageCluster := &ocsv1.StorageCluster{}
```

Comment:
What about doing `relatedObject = &ocsv1.StorageCluster{}` instead, would a `Get()` on `relatedObject` still succeed? I forget if that's valid in go...

Reply:
That won't be able to hold flashSystemCluster.